workflow-core/src/main/java/com/squareup/workflow1/LifecycleWorker.kt

```diff
@@ -73,5 +73,7 @@ abstract class LifecycleWorker : Worker<Nothing> {
   /**
    * Equates [LifecycleWorker]s that have the same concrete class.
    */
-  override fun doesSameWorkAs(otherWorker: Worker<*>): Boolean = true
+  override fun doesSameWorkAs(otherWorker: Worker<*>): Boolean {
+    return otherWorker::class == this::class
```

Comment:
My only concern with this is that it's a different default than other workers get. But I don't know if that's actually a valid concern. The usual "when in doubt, use keys" rule still applies.

Reply:
I think the difference makes sense here. I can't think of any other workers that we customize via subclassing this way.